platform/ext_bitbucket.go

```diff
@@ -0,0 +1,216 @@
+	var result string
+	if len(annotations) == 0 {
+		result = bitBucketReportPassed
+	} else {
+		result = bitBucketReportFailed
+	}
+
+	data := bbapi.NewReport()
+	data.SetTitle(toolName + " ")
+	data.SetReportType(bitBucketReportType)
+	data.SetReporter(bitBucketReporter)
+	data.SetLogoUrl(bitBucketAvatar)
+	data.SetLink(cloudUrl)
+	data.SetDetails(getProblemsFoundMessage(len(annotations)))
+	data.SetResult(result)
+	return *data
+}
+
+// buildAnnotation builds an annotation to be sent to BitBucket Code Insights
+func buildAnnotation(result *sarif.Result, ruleDescription string, reportLink string) bbapi.ReportAnnotation {
+	bbSeverity, ok := toBitBucketSeverity[getSeverity(result)]
+	if !ok {
+		log.Debugf("Unknown SARIF severity: %s", getSeverity(result))
+		bbSeverity = bitBucketLow
+	}
+	location := result.Locations[0].PhysicalLocation
+
+	data := bbapi.NewReportAnnotation()
+	data.SetExternalId(getFingerprint(result))
+	data.SetAnnotationType(bitBucketAnnotationType)
+	data.SetSummary(fmt.Sprintf("%s: %s", result.RuleId, result.Message.Text))
+	data.SetDetails(ruleDescription)
+	data.SetSeverity(bbSeverity)
+	data.SetLine(int32(location.Region.StartLine))
+	data.SetPath(location.ArtifactLocation.Uri)
+	data.SetLink(reportLink)
+	return *data
+}
+
+// getBitBucketClient returns a BitBucket API client with proper configuration by bbapi package
+func getBitBucketClient() *bbapi.APIClient {
+	config := bbapi.NewConfiguration()
+	config.HTTPClient = &http.Client{
+		Timeout: httpTimeout,
+	}
+	server := bbapi.ServerConfiguration{
+		URL:         "https://api.bitbucket.org/2.0",
+		Description: `HTTPS API endpoint`,
+	}
+	if isBitBucket() {
+		var proxyURL *url.URL
+		if isBitBucketPipe() {
+			proxyURL, _ = url.Parse(pipeProxyURL)
+		} else {
+			proxyURL, _ = url.Parse(pipelineProxyURL)
+		}
+		config.HTTPClient.Transport = &http.Transport{
+			Proxy: http.ProxyURL(proxyURL),
+		}
+		//goland:noinspection HttpUrlsUsage
+		server = bbapi.ServerConfiguration{
+			URL: "http://api.bitbucket.org/2.0",
+		}
+	}
+	config.Servers = bbapi.ServerConfigurations{server}
+	return bbapi.NewAPIClient(config)
+}
+
+// UnexpectedResponseError is triggered when we have unexpected response from CodeInsights API
+type UnexpectedResponseError struct {
+	Code int
+	Body []byte
+}
+
+func (e UnexpectedResponseError) Error() string {
+	msg := fmt.Sprintf("received unexpected %d code from Bitbucket API", e.Code)
+	if len(e.Body) > 0 {
+		msg += " with message:\n" + string(e.Body)
+	}
+	return msg
+}
+
+// checkBitBucketApiError checks if the API call was successful
+func checkBitBucketApiError(err error, resp *http.Response, expectedCode int) error {
+	if err != nil {
+		return fmt.Errorf("bitbucket Cloud API error: %w", err)
+	}
+	if resp != nil && resp.StatusCode != expectedCode {
+		body, _ := io.ReadAll(resp.Body)
+		return UnexpectedResponseError{
+			Code: resp.StatusCode,
+			Body: body,
```

Comment:
I would not log response bodies by default, can we move that to debug level instead, and not propagate it with the new error?